src/Core/ILoggerExtensions.cs

```diff
@@ -45,5 +49,19 @@ public static void LogDebug(this ILogger logger, string message, params object[]
                 logger.WriteLine("DEBUG: " + text);
             }
         }
+
+        /// <summary>
+        /// Extended debug logging that includes file, caller, thread and timestamp.
+        public static void LogDebugExtended(this ILogger logger, string message, [CallerFilePath] string callerFilePath = null, [CallerMemberName] string callerMemberName = null)
```

Comment:
Additional helper method with more information by default.
I didn't change the existing method because the way it is used is slightly different: the callers all follow a particular pattern and log with a friendly prefix and the additional info they require.
Changing the existing method would have made those messages too verbose.